packages/integration-karma/test/shadow-dom/Event-properties/Event.target.spec.js

```diff
@@ -55,3 +56,28 @@ if (!process.env.NATIVE_SHADOW) {
         });
     });
 }
+
+describe('When event target is accessed async', () => {
+    if (!process.env.NATIVE_SHADOW) {
+        it('#W-6586380 - should return the original target, if original target node is not keyed', done => {
+            spyOn(console, 'error');
+            const elm = createElement('x-parent-with-dynamic-child', {
+                is: XParentWithDynamicChild,
+            });
+            document.body.appendChild(elm);
+            const child = elm.shadowRoot.querySelector('x-child-with-out-lwc-dom-manual');
+            const originalTarget = child.shadowRoot.querySelector('span');
+            elm.eventListener = evt => {
+                expect(evt.currentTarget).toBe(elm.shadowRoot.querySelector('div'));
+                expect(evt.target).toBe(child);
+                setTimeout(() => {
+                    // Expect event target to be not retargeted
+                    expect(evt.currentTarget).toBeNull();
+                    expect(evt.target).toBe(originalTarget);
+                    done();
+                });
+            };
+            originalTarget.click();
+        });
```

Comment:
This test can pass without the `eventListener` being invoked. I'd suggest adding a failing condition if that occurs in order to guarantee that it actually executes the assertion blocks inside the `eventListener` function.

Reply:
I believe it will fail because it has no assertions if that method is never called.

Reply:
The test will fail if the `done()` in Line 77 is not called. This will ensure the test passes only if event listener is invoked.